PublicMapping/districtbuilder
# Show / remove reference layers

## Overview

This has two main features:
 - Shows reference layers that have been added on the map (
 - Allows deleting existing reference layers

Also includes some tweaks to what was added in #902
 - Allows not setting the label field when adding a reference layer
 - Hides the reference layer section when the metrics are expanded

### Checklist


### Demo

Showing point/polygon layers:
![Toggle](https://user-images.githubusercontent.com/4432106/132064371-a91d9b27-90a7-49b4-bca0-30fbb6b02b81.gif)

Deleting layers:
![Delete](https://user-images.githubusercontent.com/4432106/132064383-3ce21f6c-9af2-4af2-bb18-179f0901168e.gif)

Label option no longer required:
![NoLabels](https://user-images.githubusercontent.com/4432106/132064420-70d7bd32-25e5-4626-971c-691f082f8509.gif)


## Testing Instructions

- `scripts/update`
- Expand the reference layers section & add some CSV / geojson reference layers (I found state open data portals to be helpful resources when testing)
- Toggle the layers on/off using the checkboxes
- Try removing a reference layer using the "..." menu
- Reference layers should be visible when looking at another users map, but the "..." menu should not be

Closes #862 
Closes #863
